contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -163,26 +154,36 @@ contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, AutomationComp
     return targetsToFund;
   }
 
-  /// @notice Send funds to the targets provided.
-  /// @param targetAddresses the list of targets to fund
   function topUp(address[] memory targetAddresses) public whenNotPaused {
-    uint256 topUpAmount = s_topUpAmount;
-    uint256 stopIdx = targetAddresses.length;
-    uint256 numCanFund = LINK_TOKEN.balanceOf(address(this)) / topUpAmount;
-    stopIdx = numCanFund < stopIdx ? numCanFund : stopIdx;
-    for (uint256 idx = 0; idx < stopIdx; idx++) {
-      (bool exists, uint256 minBalance) = s_watchList.tryGet(targetAddresses[idx]);
-      if (!exists) {
-        emit TopUpBlocked(targetAddresses[idx]);
+    uint256 localBalance = LINK_TOKEN.balanceOf(address(this));
+    Target memory target;
+    address[] memory targetsTopUp = new address[](targetAddresses.length);
+    for (uint256 idx = 0; idx < targetAddresses.length; idx++) {
+      target = s_targets[targetAddresses[idx]];
+      if (localBalance >= target.topUpAmount) {
```

Comment:
inline to 
`if (localBalance >= s_targets[targetAddresses[idx]].topUpAmount) {`